### DISCUSSION TITLE
BIGINT vs UUID vs ULID performance

### DISCUSSION THREAD

**Discussion by ghost** [dylanhart/ulid-rs#47] [General]
Thank you for your work on this library.

Would you have any feedback on the performance difference in Postgres DB for primary key on a table with 1 million records for read, write sorting between:

BIGINT
UUID v7
ULID

Particularly on multiple table join queries with either of above as PK and FK

**Comment by dylanhart** [comment]
I don't have any experience with Postgres. Generally whatever is handled natively by your database should be the way to go. I'd prefer storing ulid values in numeric format since there are two bits of ambiguity in the string format (130 bits for base32 string, 128 bits for integer). UUID v7 looks like a good spec!
